CMakeLists.txt

```diff
@@ -267,13 +267,16 @@ target_link_libraries(torrent-rasterbar ${Boost_LIBRARIES} ${CMAKE_THREAD_LIBS_I
 #add_definitions(-DBOOST_ASIO_HASH_MAP_BUCKETS=1021  -D__USE_W32_SOCKETS -DWIN32_LEAN_AND_MEAN )
 
 if (WIN32)
-	target_link_libraries(torrent-rasterbar wsock32 ws2_32 Iphlpapi)
+	target_link_libraries(torrent-rasterbar wsock32 ws2_32 Iphlpapi dbghelp)
```

Comment:
``dbghlp`` is only needed for certain kinds of builds. when asserts are enabled or asio-debugging. Is it OK to add a dependency on it unconditionally here?
I know some systems may not actually have dbghlp installed (wine for instance). But maybe all of the dynamic library dependencies are loaded lazily anyway so it would never be a problem

Reply:
I check this and let know

Reply:
you are right, dbghelp is only needed during debug build, but cmake doesn't provide `FindDbgHelp.cmake` module, we can use:
```
target_link_libraries(torrent-rasterbar debug dbghelp)

Reply:
I would kind of expect the linker to detect that there aren't any references to the library and just drop it.

Reply:
you are right, dbghelp is only needed during debug build, but cmake doesn't provide `FindDbgHelp.cmake` module, we can use:
```
target_link_libraries(torrent-rasterbar debug dbghelp)
```
or add `FindDbgHelp.cmake` too.
What do you mean?
The other thing is, that I can build on master without this lib :/ How it's possible :)

Reply:
perhaps "dbghelp.h" contains a "#pragma lib" directive to link against it or something like that

Reply:
I think that asserts are disabled for debug builds too in master, `TORRENT_USE_ASSERTS` is not defined, i'm debugging this now. 
As you can see here https://github.com/arvidn/libtorrent/blob/master/include/libtorrent/config.hpp#L531, even the comment message says something else and that is the reason, why libtorrent builds without dbghelp, but i'm not 100% sure now.

Reply:
ah, yes. The logic to enable asserts in debug mode was moved to the Jamfile, but not into the CMakeList.txt iirc. ``TORRENT_USE_ASSERTS`` should be defined when building in debug mode

Reply:
ok, thx